packages/vue-router/src/router.ts

```diff
@@ -212,59 +221,77 @@ export const createIonRouter = (opts: IonicVueRouterOptions, router: Router) =>
           }
         }
       }
+      nextRouteInfo = {
+        ...incomingRouteParams,
+        id: generateId('routeInfo'),
+        replacedRoute: undefined,//reset to undefined
+        pathname: location.path,
+        search: location.fullPath && location.fullPath.split('?')[1] || '',
+        params: location.params && location.params,
+      }
 
-      let routeInfo: RouteInfo;
-      if (incomingRouteParams?.id) {
-        routeInfo = {
-          ...incomingRouteParams,
-          lastPathname: leavingLocationInfo.pathname
-        }
-
-      } else {
-        const isPushed = incomingRouteParams.routerAction === 'push' && incomingRouteParams.routerDirection === 'forward';
-        routeInfo = {
-          id: generateId('routeInfo'),
-          ...incomingRouteParams,
-          lastPathname: leavingLocationInfo.pathname,
-          pathname: location.path,
-          search: location.fullPath && location.fullPath.split('?')[1] || '',
-          params: location.params && location.params,
-          prevRouteLastPathname: leavingLocationInfo.lastPathname
-        }
+      const isNewTab = nextRouteInfo.tab !== leavingRouteInfo.tab;
+      const isPushed = incomingRouteParams.routerAction === 'push' && incomingRouteParams.routerDirection === 'forward';
+      if (isPushed) {
+        /**
+         * Push/Forward occurs whenever navigating forward, page -> page/a -> page/b
+         *
+         * Going to page/a and page/b would be push events
+         */
+        nextRouteInfo.lastPathname = leavingRouteInfo.pathname;
+        nextRouteInfo.tab = leavingRouteInfo.tab;
+        //@TODO: Check if bug, why would pathname be blank?
+        nextRouteInfo.pushedByRoute = (leavingRouteInfo.pathname !== '') ? leavingRouteInfo.pathname : undefined;
```

Comment:
I believe you can create an instance where `pathname` is `''`, when using `history.pushState`/`history.replaceState`, but also I cannot immediately think of an instance where that's appropriate. From the docs, it appears it happens when there is no path (https://developer.mozilla.org/en-US/docs/Web/API/Location/pathname).

Reply:
I'll add this comment to explain the blank scenario in code

Reply:
I saw the first route has path = '' before we redirect it to the desired page too. Also I did break tabs so I will fix that and remove commented out code and add test cases for my bug scenario and the tab scenario. Pretty sure there are still bugs when navigating around with push,replace,go,back in complex scenarios so I will see what I find. 


@sean-perkins When going between tabs on the same `ion-tab-bar` do we want each new tab to be pushed onto locationHistory or should the tab replace the previous tab if they keep clicking between tabs? The code base seems to handle tabs in unique ways and I was wondering if we could just have them behave as replacements, that should simplify things.